Src/Concrete/Base.hs

```diff
@@ -134,27 +135,68 @@ data ExtractMode
   | InterestingExtract
   deriving (Show, Eq)
 
-data Actor jd ch bd av syn var tm pat cnnct stk
- = Branch Range (Actor jd ch bd av syn var tm pat cnnct stk) (Actor jd ch bd av syn var tm pat cnnct stk)
- | Spawn Range ExtractMode jd ch (Actor jd ch bd av syn var tm pat cnnct stk)
- | Send Range ch tm (Actor jd ch bd av syn var tm pat cnnct stk)
- | Recv Range ch (bd, Actor jd ch bd av syn var tm pat cnnct stk)
- | Connect Range cnnct
- | Note Range (Actor jd ch bd av syn var tm pat cnnct stk)
- | FreshMeta Range syn (av, Actor jd ch bd av syn var tm pat cnnct stk)
- | Under Range (Scope Variable (Actor jd ch bd av syn var tm pat cnnct stk))
- | Match Range tm [(pat, Actor jd ch bd av syn var tm pat cnnct stk)]
+data Phase = Concrete | Abstract
+
+type family JUDGEMENTFORM (ph :: Phase) :: *
+type family CHANNEL (ph :: Phase) :: *
+type family BINDER (ph :: Phase) :: *
+type family ACTORVAR (ph :: Phase) :: *
+type family SYNTAXDESC (ph :: Phase) :: *
+type family TERMVAR (ph :: Phase) :: *
+type family TERM (ph :: Phase) :: *
+type family PATTERN (ph :: Phase) :: *
+type family CONNECT (ph :: Phase) :: *
+type family STACK (ph :: Phase) :: *
+type family STACKDESC (ph :: Phase) :: *
+
+type instance JUDGEMENTFORM Concrete = Variable
+type instance CHANNEL Concrete = Variable
+type instance BINDER Concrete = RawP
+type instance ACTORVAR Concrete = Variable
+type instance SYNTAXDESC Concrete = Raw
+type instance TERMVAR Concrete = Variable
+type instance TERM Concrete = Raw
+type instance PATTERN Concrete = RawP
+type instance CONNECT Concrete = CConnect
+type instance STACK Concrete = Variable
+type instance STACKDESC Concrete = ()
+
+type FORMAT (ph :: Phase) = [Format Directive Debug (TERM ph)]
+
+data ACTOR (ph :: Phase)
```

Comment:
I couldn't face the prospect of adding yet another parameter to this datatype so I went ahead and did the `Phase` refactoring.